pkg/util/goroutinemap/goroutinemap_test.go

```diff
@@ -106,9 +153,63 @@ func Test_NewGoRoutineMap_Positive_SecondOpAfterFirstPanics(t *testing.T) {
 	}
 }
 
+func Test_NewGoRoutineMap_Positive_SecondOpAfterFirstPanicsWithExpBackoff(t *testing.T) {
+	// Arrange
+	grm := NewGoRoutineMap(true /* exponentialBackOffOnError */)
+	operationName := "operation-name"
+	operation1 := generatePanicFunc()
+	err1 := grm.Run(operationName, operation1)
+	if err1 != nil {
+		t.Fatalf("NewGoRoutine failed. Expected: <no error> Actual: <%v>", err1)
+	}
+	operation2 := generateNoopFunc()
+
+	// Act
+	err2 := retryWithExponentialBackOff(
+		time.Duration(500*time.Millisecond), // Longer duration to accommodate for backoff
```

Comment:
Moved to consts. Will leave it at 500, tested values higher and lower, and 500 results in the fastest test times.